### PR TITLE
🎁 Add loading splash screen and speed up redirect

### PR DESCRIPTION
**Pull request by kirkkwang**
The clients requested a loading screen that covers the entire page so users can't click on anything while the redirect is happening.

https://github.com/scientist-softserv/palni-palci/assets/19597776/9efc588f-eaff-41f1-a3c6-0380da8a9514

### CODE REVIEW

File: app/models/cdl.rb [hunk 1]
```diff
@@ -50,4 +55,24 @@ class Cdl < ActiveFedora::Base
   # including `include ::Hyrax::BasicMetadata`. All properties must
   # be declared before their values can be ordered.
   include OrderMetadataValues
+
+  private
+
+    def create_group_and_add_members
+      return if is_child
+
+      # IiifPrint::Jobs::CreateRelationshipsJob has a 10 min timeout
```
Comment (jeremyf) [inline app/models/cdl.rb:64]: Wondering about the timing of this?  Should we instead key off of the `CreateRelationshipsJob`?  We'll need to talk this over.

**Review by jeremyf — CHANGES_REQUESTED** [review]

File: app/jobs/create_group_and_add_members_job.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+  def perform(cdl_id, retries = 0)
+    work = Cdl.where(id: cdl_id).first
+    return if work.nil?
+
+    page_count = work.file_sets.first.page_count.first.to_i
+    child_model = work.iiif_print_config.pdf_split_child_model
+    child_works_count = work.members.select { |member| member.is_a?(child_model) }.count
+
+    if page_count == child_works_count
+      group = Hyrax::Group.create(name: work.id)
```
Comment (jeremyf) [inline app/jobs/create_group_and_add_members_job.rb:25]: Given that we have retry attempts favoring a find or create will allow us to:

1. Avoid name collisions
2. Allow for greater fault tolerance.

```suggestion
      group = Hyrax::Group.find_or_create_by!(name: work.id)
```

File: app/models/cdl.rb (hunk 1 above)
Comment (jeremyf) [reply app/models/cdl.rb:64]: I think we should decorate the `IiifPrint::Jobs::CreateRelationshipsJob` to conditionally launch this job.

Comment (kirkkwang) [reply app/models/cdl.rb:64]: but the job itself has a check built in as well as retries if conditions aren't met, but I do agree with keying off of `CreateRelationshipsJob`.

File: app/jobs/create_group_and_add_members_job.rb [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        assign_read_groups(member, group.name)
+      end
+
+      work.save
+      group.save
+    else
+      return if retries > RETRY_MAX
+
+      retries += 1
+      CreateGroupAndAddMembersJob.set(wait: 10.minutes).perform_later(cdl_id, retries)
```
Comment (jeremyf) [inline app/jobs/create_group_and_add_members_job.rb:37]: Yes this one gets to retry!

**Review by jeremyf — APPROVED** [review]

**Pull request merged**